### PR TITLE
logging

### PR DESCRIPTION
**Pull request by ALERTua**
It's quite boring to watch the container empty log output, so I expanded the logging.
With these changes, the logging looks like this:
<details>
<summary>Click to expand</summary>
Subscribed to topic qingping/582D34826EAC/up

Successfully connected to MQTT

Unknown data key: 67

Publishing qingping/582D34826EAC/parsed/productId: "3\u0000"

Publishing qingping/582D34826EAC/parsed/firmware_version: "2.0.0"

Publishing qingping/582D34826EAC/parsed/hardwareVersion: "0000"

Publishing qingping/582D34826EAC/parsed/isPluggedInToPower: true

Publishing qingping/582D34826EAC/parsed/co2IsBeingCalibrated: false

Publishing qingping/582D34826EAC/parsed/wirelessModuleFirmwareVersion: "1.9.5"

Publishing qingping/582D34826EAC/parsed/mcuFirmwareVersion: "2.0.0"

Publishing qingping/582D34826EAC/parsed/isGoingIntoLowPowerMode: true

Received message with timestamp 1704977505

Publishing qingping/582D34826EAC/parsed/timestamp: 1704977505

Published discovery data to topic homeassistant/sensor/582D34826EAC_temperature/config: {"device":{"identifiers":["582D34826EAC"],"name":"582D34826EAC","manufacturer":"QingPing","model":"CO2 & Temp & RH Monitor"},"device_class":"temperature","state_topic":"qingping/582D34826EAC/parsed/temperature","unique_id":"582D34826EAC_temperature","name":"582D34826EAC temperature","unit_of_measurement":"°C"}

Publishing qingping/582D34826EAC/parsed/temperature: 21.2

Published discovery data to topic homeassistant/sensor/582D34826EAC_humidity/config: {"device":{"identifiers":["582D34826EAC"],"name":"582D34826EAC","manufacturer":"QingPing","model":"CO2 & Temp & RH Monitor"},"device_class":"humidity","state_topic":"qingping/582D34826EAC/parsed/humidity","unique_id":"582D34826EAC_humidity","name":"582D34826EAC humidity","unit_of_measurement":"%"}

Publishing qingping/582D34826EAC/parsed/humidity: 37.6

Published discovery data to topic homeassistant/sensor/582D34826EAC_co2_ppm/config: {"device":{"identifiers":["582D34826EAC"],"name":"582D34826EAC","manufacturer":"QingPing","model":"CO2 & Temp & RH Monitor"},"device_class":"carbon_dioxide","state_topic":"qingping/582D34826EAC/parsed/co2_ppm","unique_id":"582D34826EAC_co2_ppm","name":"582D34826EAC co2_ppm","unit_of_measurement":"ppm"}

Publishing qingping/582D34826EAC/parsed/co2_ppm: 541

Published discovery data to topic homeassistant/sensor/582D34826EAC_battery/config: {"device":{"identifiers":["582D34826EAC"],"name":"582D34826EAC","manufacturer":"QingPing","model":"CO2 & Temp & RH Monitor"},"device_class":"battery","state_topic":"qingping/582D34826EAC/parsed/battery","unique_id":"582D34826EAC_battery","name":"582D34826EAC battery","unit_of_measurement":"%"}

Publishing qingping/582D34826EAC/parsed/battery: 100

Publishing qingping/582D34826EAC/parsed/productId: "3\u0000"

Publishing qingping/582D34826EAC/parsed/isGoingIntoLowPowerMode: true

Publishing qingping/582D34826EAC/parsed/historicalData: "54 e2 9f 65 3c 0 78 61 2c 2f 2 64 7a 61 2c 2b 2 64 7b 61 2c 25 2 64 7b 71 2c 21 2 64 79 71 2c 1f 2 64 79 81 2c 1c 2 64 78 81 2c 19 2 64 76 81 2c 16 2 64 75 81 2c 10 2 64"

Unknown data key: 44

Publishing qingping/582D34826EAC/parsed/uploadDataInterval: 600

Publishing qingping/582D34826EAC/parsed/recordDataInterval: 60

Publishing qingping/582D34826EAC/parsed/undocumentedValue: "232,3"

Publishing qingping/582D34826EAC/parsed/temperatureUnit: "celsius"

Publishing qingping/582D34826EAC/parsed/temperatureOffset: 6551.6

Publishing qingping/582D34826EAC/parsed/temperatureOffsetPercentage: 0

Publishing qingping/582D34826EAC/parsed/humidityOffset: 5

Publishing qingping/582D34826EAC/parsed/humidityOffsetPercentage: 0

Publishing qingping/582D34826EAC/parsed/productId: "3\u0000"

Publishing qingping/582D34826EAC/parsed/co2MeasurementInterval: 1800

Publishing qingping/582D34826EAC/parsed/autoOffTime: 0

Publishing qingping/582D34826EAC/parsed/timeMode: "24h"

Publishing qingping/582D34826EAC/parsed/co2OffsetPercentage: 0

Publishing qingping/582D34826EAC/parsed/co2Offset: 0

Publishing qingping/582D34826EAC/parsed/co2ASC: true

Publishing qingping/582D34826EAC/parsed/co2IsBeingCalibrated: false

Unknown data key: 67

Publishing qingping/582D34826EAC/parsed/productId: "3\u0000"

Publishing qingping/582D34826EAC/parsed/firmware_version: "2.0.0"

Publishing qingping/582D34826EAC/parsed/hardwareVersion: "0000"

Publishing qingping/582D34826EAC/parsed/isPluggedInToPower: true

Publishing qingping/582D34826EAC/parsed/co2IsBeingCalibrated: false

Publishing qingping/582D34826EAC/parsed/wirelessModuleFirmwareVersion: "1.9.5"

Publishing qingping/582D34826EAC/parsed/mcuFirmwareVersion: "2.0.0"

Publishing qingping/582D34826EAC/parsed/isGoingIntoLowPowerMode: true

Received message with timestamp 1704977541

Publishing qingping/582D34826EAC/parsed/timestamp: 1704977541

Published discovery data to topic homeassistant/sensor/582D34826EAC_temperature/config: {"device":{"identifiers":["582D34826EAC"],"name":"582D34826EAC","manufacturer":"QingPing","model":"CO2 & Temp & RH Monitor"},"device_class":"temperature","state_topic":"qingping/582D34826EAC/parsed/temperature","unique_id":"582D34826EAC_temperature","name":"582D34826EAC temperature","unit_of_measurement":"°C"}

Publishing qingping/582D34826EAC/parsed/temperature: 21.2

Published discovery data to topic homeassistant/sensor/582D34826EAC_humidity/config: {"device":{"identifiers":["582D34826EAC"],"name":"582D34826EAC","manufacturer":"QingPing","model":"CO2 & Temp & RH Monitor"},"device_class":"humidity","state_topic":"qingping/582D34826EAC/parsed/humidity","unique_id":"582D34826EAC_humidity","name":"582D34826EAC humidity","unit_of_measurement":"%"}

Publishing qingping/582D34826EAC/parsed/humidity: 37.8

Published discovery data to topic homeassistant/sensor/582D34826EAC_co2_ppm/config: {"device":{"identifiers":["582D34826EAC"],"name":"582D34826EAC","manufacturer":"QingPing","model":"CO2 & Temp & RH Monitor"},"device_class":"carbon_dioxide","state_topic":"qingping/582D34826EAC/parsed/co2_ppm","unique_id":"582D34826EAC_co2_ppm","name":"582D34826EAC co2_ppm","unit_of_measurement":"ppm"}

Publishing qingping/582D34826EAC/parsed/co2_ppm: 523

Published discovery data to topic homeassistant/sensor/582D34826EAC_battery/config: {"device":{"identifiers":["582D34826EAC"],"name":"582D34826EAC","manufacturer":"QingPing","model":"CO2 & Temp & RH Monitor"},"device_class":"battery","state_topic":"qingping/582D34826EAC/parsed/battery","unique_id":"582D34826EAC_battery","name":"582D34826EAC battery","unit_of_measurement":"%"}

Publishing qingping/582D34826EAC/parsed/battery: 100

Publishing qingping/582D34826EAC/parsed/productId: "3\u0000"

Publishing qingping/582D34826EAC/parsed/isGoingIntoLowPowerMode: true

Publishing qingping/582D34826EAC/parsed/historicalData: "70 e4 9f 65 3c 0 78 91 2c 16 2 64"
</details>

At least you can say it's working now :)

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -23,10 +23,12 @@ mqttClient.on("connect", () => {
   console.log("Successfully connected to MQTT");
 });
 
-mqttClient.subscribe(Env.SENSOR_TOPIC);
+const sensorTopic = Env.SENSOR_TOPIC;
```
Comment (niklasarnitz) [inline index.ts]: This assignment is not needed.
`Env.SENSOR_TOPIC` is already a defined constant.

Comment (niklasarnitz) [inline index.ts]:
```suggestion
```

File: index.ts [hunk 2]
```diff
@@ -23,10 +23,12 @@ mqttClient.on("connect", () => {
...
+mqttClient.subscribe(sensorTopic);
```
Comment (niklasarnitz) [inline index.ts]:
```suggestion
mqttClient.subscribe(Env.SENSOR_TOPIC);
```

File: index.ts [hunk 3]
```diff
@@ -23,10 +23,12 @@ mqttClient.on("connect", () => {
...
+console.log(`Subscribed to topic ${sensorTopic}`);
```
Comment (niklasarnitz) [inline index.ts]:
```suggestion
console.log(`Subscribed to topic ${Env.SENSOR_TOPIC}`);
```

File: index.ts [hunk 4]
```diff
@@ -23,10 +23,12 @@ mqttClient.on("connect", () => {
...
 
 mqttClient.on("message", (topic, message) => {
-  if (topic === Env.SENSOR_TOPIC) {
+  if (topic === sensorTopic) {
```
Comment (niklasarnitz) [inline index.ts]:
```suggestion
  if (topic === Env.SENSOR_TOPIC) {
```

File: index.ts [hunk 5]
```diff
@@ -46,16 +48,22 @@ mqttClient.on("message", (topic, message) => {
         const discoveryData = getHomeAssistantDiscoveryJson(key);
 
         if (discoveryData) {
+          const topic = getHomeAssistantMQTTTopic(key);
+          const stringifiedData = JSON.stringify(discoveryData);
```
Comment (niklasarnitz) [inline index.ts]:
```suggestion
          const stringifiedDiscoveryData = JSON.stringify(discoveryData);
```

File: index.ts [hunk 6]
```diff
@@ -46,16 +48,22 @@ mqttClient.on("message", (topic, message) => {
...
           mqttClient.publish(
-            getHomeAssistantMQTTTopic(key),
-            JSON.stringify(discoveryData)
+            topic,
+            stringifiedData
```
Comment (niklasarnitz) [inline index.ts]:
```suggestion
            stringifiedDiscoveryData
```

File: index.ts [hunk 7]
```diff
@@ -46,16 +48,22 @@ mqttClient.on("message", (topic, message) => {
...
           );
+          console.log(`Published discovery data to topic ${topic}: ${stringifiedData}`);
```
Comment (niklasarnitz) [inline index.ts]:
```suggestion
          console.log(`Published discovery data to topic ${topic}: ${stringifiedDiscoveryData}`);
```

**Review by niklasarnitz — CHANGES_REQUESTED** [review]
Very good addition, thank you.

Please have a look at my suggestions.

**Comment by niklasarnitz** [comment]
The only question is if every datapoint needs to be logged.
I would probably want a flag to toggle between:
`Published packet to topic: ${topic}` and `Published packet to topic ${topic}: ${data}`

What do you think?

**Comment by ALERTua** [comment]
I don't see any harm in spamming the log with not only the topics but also the data. One can debug their experiments more easily with this. I think it's ok for now. We can wrap these prints into a more complex logger in the future if needed.

**Pull request merged**